lib/dotcom/utils/service_date_time.ex

```diff
@@ -0,0 +1,224 @@
+defmodule Dotcom.Utils.ServiceDateTime do
+  @moduledoc """
+  A collection of functions that helps to work with date_times with regard to service times.
+  Currently, we consider the most general case where service ends at 02:59:59 and starts at 03:00:00.
+
+  In the future, we aim to add route-specific service times.
+
+  Note that functions like `service_today?/1` and `service_this_week?/1` are not exlusive.
+  A date_time can have service today and service this week.
+  I.e., today is both today and this week.
+
+  However, the `service_range/1` function *is* exclusive.
+  Given a date_time in today's service range, it will return :today.
+  """
+
+  require Logger
+
+  use Timex
+
+  import Dotcom.Utils.DateTime, only: [coerce_ambiguous_time: 1, now: 0, timezone: 0]
+
+  alias Dotcom.Utils
+
+  @type service_range() :: :past | :today | :this_week | :next_week | :later
+
+  @service_rollover_time Application.compile_env!(:dotcom, :service_rollover_time)
+  @timezone timezone()
+
+  @doc """
+  Returns the time at which service rolls over from 'today' to 'tomorrow'.
+  """
+  def service_rollover_time(), do: @service_rollover_time
+
+  @doc """
+  Get the service date for the given date_time.
+  If the time is before 3am, we consider it to be the previous day.
+  """
+  @spec service_date() :: Date.t()
+  @spec service_date(DateTime.t()) :: Date.t()
+  def service_date(date_time \\ now()) do
+    if date_time.hour < @service_rollover_time.hour do
+      Timex.shift(date_time, hours: -@service_rollover_time.hour) |> Timex.to_date()
```

Comment:
Would we need `corece_ambiguous_time/1` here too?

Reply:
Yeah, good find. Super annoying, but we have to call it after every shift.